configapi/api_inventory.go

```diff
@@ -274,20 +273,64 @@ func handlePostUpf(c *gin.Context) error {
 	return nil
 }
 
-func handleDeleteUpf(c *gin.Context) error {
-	upfHostname, exists := c.Params.Get("upf-hostname")
-	if !exists {
-		errorMessage := "delete UPF request is missing upf-hostname"
-		logger.ConfigLog.Errorln(errorMessage)
-		return fmt.Errorf("%s", errorMessage)
+func updateGnbInNetworkSlices(gnbName string) error {
+	filterByGnb := bson.M{
+		"site-info.gNodeBs": bson.M{
+			"$elemMatch": bson.M{"name": gnbName},
+		},
 	}
-	logger.ConfigLog.Infof("received delete UPF %v", upfHostname)
-	msg := configmodels.ConfigMessage{
-		MsgType:     configmodels.Inventory,
-		MsgMethod:   configmodels.Delete_op,
-		UpfHostname: upfHostname,
+	rawNetworkSlices, errGetMany := dbadapter.CommonDBClient.RestfulAPIGetMany(sliceDataColl, filterByGnb)
+	if errGetMany != nil {
+		logger.DbLog.Errorln("failed to fetch network slices:", errGetMany)
+		return errGetMany
+	}
+	for _, rawNetworkSlice := range rawNetworkSlices {
+		var networkSlice configmodels.Slice
+		if err := json.Unmarshal(configmodels.MapToByte(rawNetworkSlice), &networkSlice); err != nil {
+			return err
+		}
+		filteredGNodeBs := []configmodels.SliceSiteInfoGNodeBs{}
+		for _, gnb := range networkSlice.SiteInfo.GNodeBs {
+			if gnb.Name != gnbName {
+				filteredGNodeBs = append(filteredGNodeBs, gnb)
+			}
+		}
+		filteredGNodeBsJSON, err := json.Marshal(filteredGNodeBs)
+		if err != nil {
+			logger.DbLog.Errorln("error marshaling GNodeBs: %v", err)
+			continue
+		}
+		patchJSON := []byte(
+			fmt.Sprintf(`[{"op": "replace", "path": "/site-info/gNodeBs", "value": %s}]`,
+				string(filteredGNodeBsJSON)),
+		)
+		filterBySliceName := bson.M{"slice-name": networkSlice.SliceName}
+		err = dbadapter.CommonDBClient.RestfulAPIJSONPatch(sliceDataColl, filterBySliceName, patchJSON)
+		if err != nil {
+			logger.DbLog.Warnln("failed to update network slice:", networkSlice.SliceName, "Error:", err)
+		}
+	}
+	return nil
+}
+
+func updateUpfInNetworkSlices(hostname string, patchJSON []byte) error {
+	filterByUpf := bson.M{"site-info.upf.upf-name": hostname}
+	rawNetworkSlices, errGetMany := dbadapter.CommonDBClient.RestfulAPIGetMany(sliceDataColl, filterByUpf)
+	if errGetMany != nil {
+		logger.DbLog.Errorln("failed to fetch network slices:", errGetMany)
```

Comment:
```suggestion
		logger.DbLog.Errorf("failed to fetch network slices: %v", errGetMany)
```
Using Errorf to format log messages.